packages/cli/src/services/redis-client.service.ts

```diff
@@ -156,4 +190,33 @@ export class RedisClientService {
 				return { host, port: parseInt(port) };
 			});
 	}
+
+	@Debounce(1000)
```

Comment:
Why do we need the debounce here?

Reply:
Debouncing prevents all six clients spamming the logs with their calls to `retryStrategy`, i.e. we consolidate the logs on `connection-lost` and `connection-recovered` from six to one.

```ts
export type RedisClientType = N8nRedisClientType | BullRedisClientType;

/**
 * Redis client used by n8n.
 *
 * - `subscriber(n8n)` to listen for messages from scaling mode pubsub channels
 * - `publisher(n8n)` to send messages into scaling mode pubsub channels
 * - `cache(n8n)` for caching operations (variables, resource ownership, etc.)
 */
type N8nRedisClientType = 'subscriber(n8n)' | 'publisher(n8n)' | 'cache(n8n)';

/**
 * Redis client used internally by Bull. Suffixed with `(bull)` at `ScalingService.setupQueue`.
 *
 * - `subscriber(bull)` for event listening
 * - `client(bull)` for general queue operations
 * - `bclient(bull)` for blocking operations when processing jobs
 */
type BullRedisClientType = 'subscriber(bull)' | 'client(bull)' | 'bclient(bull)';
```